src/Dumplie/Domain/Customer/CartItem.php

```diff
@@ -36,6 +36,14 @@ public function __construct(Product $product, int $quantity)
     }
 
     /**
+     * @return Product
+     */
+    public function product() : Product
```

Comment:
Also, `CartItem` should have `::item()` instead `::product()`. Not all items in cart are `products` - eg. extended warranty is a `service` (with its own `SKU` but its not a product anymore).